Jenkinsfile

```diff
@@ -29,38 +29,33 @@ node {
         npmTag = "prerelease"
     }
 
-    lastCommitAuthor = sh (
-      script: 'git show HEAD --no-patch --format="%an"',
-      returnStdout: true
-    ).trim()
-
     lastCommitHash = sh (
       script: 'git show HEAD --no-patch --format="%h"',
       returnStdout: true
     ).trim()
 
-    if (lastCommitAuthor == "jenkins-hypothesis") {
-        echo "Skipping build of automated commit created by Jenkins"
-        return
-    }
-
     pkgName = sh (
       script: 'cat package.json | jq -r .name',
       returnStdout: true
     ).trim()
 
+    // Update local information about tags to match the remote,
+    // including removing any local tags that no longer exist.
+    //
+    // The `--prune-tags` option is not supported in Git 2.11 so we
+    // use the workaround from https://github.com/git/git/commit/97716d217c1ea00adfc64e4f6bb85c1236d661ff
+    sh "git fetch --quiet --prune origin 'refs/tags/*:refs/tags/*' "
+
+    // Determine version number for next release.
     pkgVersion = sh (
-      script: 'cat package.json | jq -r .version',
+      script: 'git tag --list | sort --version-sort --reverse | head -n1 | tail -c +2',
       returnStdout: true
     ).trim()
-
-    stage('Build') {
```

Comment:
This stage doesn't actually do anything useful. `make` on its own just prints help output. The client actually gets built during the "QA" and "prod" steps with the appropriate version number. If there is eg. a syntax error, it'll get flagged either during the Test or QA steps.